### ISSUE TITLE
[Bug]: V1 spec-decode proposer never constructs the positions buffer a both-XD-RoPE drafter seeds from

### ISSUE THREAD

**Issue by he-yufeng** [vllm-project/vllm#54555]
### 🐛 Describe the bug

The V1 spec-decode proposer crashes on the first sequential draft step for any both-XD-RoPE configuration (target model `uses_xdrope_dim > 0` and draft model `uses_xdrope_dim > 0`), because the position-buffer init and the draft-loop seed disagree about which buffer this config uses.

**Buffer init** (`vllm/v1/spec_decode/llm_base_proposer.py`, `SpecDecodeBaseProposer.__init__`):

```python
if self.uses_mrope:
    self.mrope_positions = torch.zeros(...)
elif self.uses_xdrope_dim > 0 and self.draft_uses_xdrope_dim > 0:
    self.xdrope_positions = torch.zeros(...)
else:
    self.positions = torch.zeros(...)
```

For both-XD-RoPE, `self.positions` is never constructed.

**Draft-loop seed** (`propose()`, before the `num_speculative_tokens - 1` loop):

```python
if self.uses_mrope:
    positions = self.mrope_positions[:, token_indices_to_sample]
else:
    positions = self.positions[token_indices_to_sample]
```

Only `uses_mrope` gets a dedicated branch, so a both-XD-RoPE run lands in the `else` and reads `self.positions`, which does not exist for this config: immediate `AttributeError` on the first speculative step. The same mismatch exists in `_get_positions` / `_set_positions`, which do handle the both-XD-RoPE branch, so the buffers that ARE written are never the ones the seed reads.

**Reachability.** No model in main implements `get_xdrope_input_positions` yet, so nothing in-tree can trigger this today; it is a latent crash on the landing path. The XD-RoPE machinery is clearly mid-landing (scheduler, runner, `RopeState`, #54232), the runner comment names HunYuan-VL as the intended family, and the first both-XD-RoPE drafter with `num_speculative_tokens >= 2` (sequential, non-parallel) will hit this on its very first proposal. Worth fixing before that model PR lands rather than inside it.

**Fix shape.** Treat both-XD-RoPE the way M-RoPE is treated in the seed and in `_update_positions_dependent_metadata`: seed from `self.xdrope_positions[:, token_indices_to_sample]` and take dim 0 as the slot coordinate. That is sound because XD-RoPE decode positions are the absolute token index on every dim by construction: `get_rope_state` builds the XD-RoPE state with `has_delta=False`, so `_prepare_rope_positions_kernel` uses `orig_pos` for all dims during decode. A CPU-only proposer-level test with synthetic XD-RoPE positions can pin this without a checkpoint; I am happy to open that PR if the direction sounds right.

**Relation to #54498.** Found while analyzing that issue. Different root: #54498 is a wrong-but-present coordinate for M-RoPE, this one is a missing buffer for XD-RoPE. The two fixes touch adjacent branches of the same function and should not conflict.

**Disclosure.** This analysis was done with AI assistance; every claim above was verified against current main (`810bc32`) by reading the cited code paths.

**Comment by he-yufeng** [comment]
The fix is up in #54621 with a CPU test suite for all three RoPE flavors. The seed now reads the xdrope buffer for the both-XD-RoPE config, and the slot coordinate comes from dim 0 for both multi-dim flavors, which is sound because XD-RoPE decode positions are the absolute index on every dim by construction.
